Introduce notification callbacks for the Policy Service (#433)
Microsoft/mu_basecore: PolicyServicePkg/README.md

Removes a policy from the policy list, freeing it when possible.

### _RegisterNotify_

Registers a callback that will be invoked when a policy is edited in some way. The
reasons for the callback can be filtered through the EventTypes parameter.

### _UnregisterNotify_

Removes a policy notification callback, prevent any further invocation.

## Policy Notifications

Consumers may subscribe to policy updates in multiple ways. For end-consumers

> of a policy they may use the policy GUID in their `DEPEX` for the signal policy,

Reviewer: ```suggestion
of a policy they may use the policy GUID in their `DEPEX` for the signaled finalized policy,

```

and all other consumers may use the policy notification callbacks.

### Final Policy Signal Protocol & PPI

When a policy is finalized a PPI or Protocol will be installed with the
GUID of the policy. Consumers may use this GUID to either set Protocol/PPI
notifications, or create a DEPEX dependency so that the consumer is not
dispatched until the final policy is made available. The protocol/PPI will not

> contain any useful interface and consumers are expected to use the protocol

Reviewer: ```suggestion
contain any useful interface and consumers are expected to use the Policy Service protocol

```

interface to retrieve the policy data after being notified or dispatched.

### Policy Service Callbacks

The policy service supports callbacks for various types of policy events such
as a policy being created, updated, finalized, or removed. Consumers may use the
`RegisterNotify` routine to set a callback in the event that a provided policy
undergoes the event specified in the callback registration. A given callbacks may

> receive an events with multiple events at once. For example, a finalized event will

Reviewer: ```suggestion
receive an event with multiple events at once. For example, a finalized event will

```
